src/common/async.cpp

```diff
@@ -46,6 +49,13 @@ void Async::join() {
     }
 }
 
+Async::~Async() {
+    // WARNING: This MUST be here to make sure we break the loop before we
+    // stop the thread. Not doing that leads to undefined behavior.
+    break_loop();
+    join();
```

Comment:
Ok thank you! I was wondering if the correct way of calling join was on the thread object, something like:

    if (running)
        thread.join();